### PR TITLE
Refactor routing-api cmd tests to improve readability and be more explicit about multithreading

### PR DESCRIPTION
**Pull request by andy-a-d-nguyen**
- [x] Read the [Contributing document](../blob/-/.github/CONTRIBUTING.md).

Summary
---------------
The current test setup has many helper functions defined at the bottom of test files that modify variables using closure and thereby creating side effects that are hard to follow as one reads through the tests. Additionally, there are two things about the tests that are important; a valid connection to a database and the locket executable and that the database connections are being made on separate threads than the running of the locket. The connection a locket is using to connect to a database must be ready before that locket makes the connection. In the current test setup, it is not clear how this sequential order of operation is achieved. This PR refactors the test to use `sync.WaitGroup` to explicitly declare that a working connection to a database must be ready before the locket executable is run even if these two things are run in different threads as well as replace uses of closure with function that returns values.

This PR also groups repeated values into a constants file so they can be defined once and then used in multiple places.

Backward Compatibility
---------------
Breaking Change? **No**

### CODE REVIEW

**Review by geofffranks — COMMENTED** [review]
I like the changes you've made here. I don't quite follow the addition of the waitgroup to all the database calls. What were you trying to accomplish here? Just make all the database tests run serially?

**Comment by andy-a-d-nguyen** [comment]
I tried to refactor these tests without using `sync.WaitGroup` initially and ran into panics where the locket executable was running without a valid connection to a database. I'm not sure what the mechanism allows the locket to run after a database connection is ready currently, but I think it has to do with the use of closure somehow; however, I think using closure across threads is generally not a good idea. By adding `sync.WaitGroup`, the locket will wait until a valid database connection exists when it runs.

**Comment by geofffranks** [comment]
My understanding here is that ginkgomon.Invoke() is spawning a new process using the locket binary to connect directly to the database, using the config passed in. I don't see any go routines in the dballocator.Reset() code, so i think something else is probably going on here. The waitgroup wouldn't be helping unless something was actually spawning goroutines.

 I checked out your branch, and removed the waitgroup code, and everything passed for me. Are you still able to reproduce the failures without it?

**Comment by andy-a-d-nguyen** [comment]
Strange. I removed the waitgroup and things are still passing too. Oh well, I'll update the MR with the removal. Thanks for letting me know!

**Comment by andy-a-d-nguyen** [comment]
I removed the waitgroup. Feel free to take another look. Thanks!

**Review by geofffranks — APPROVED** [review]
